Check that some model part is run on-chip

hunse opened the pull request:
Addresses #32.

tbekolay commented on nengo_loihi/tests/test_simulator.py:
```diff
@@ -0,0 +1,15 @@
+        model.config[a].on_chip = False
+
+    with pytest.raises(RuntimeError):
+        Simulator(model)
```
I'm not sure if this makes a difference (since I believe the error happens in `__init__`) but in similar tests I've been creating the simulator as a context manager here with the thought that the `__exit__` might still fire and clean things up if anything was made.

hunse replied:
Sure, I'm fine with that.

tbekolay commented on nengo_loihi/loihi_cx.py:
```diff
@@ -382,6 +382,11 @@ def get_loihi(self, seed=None):
 
+    def validate(self):
+        if len(self.cx_groups) == 0:
+            raise RuntimeError("No neurons marked for execution on-chip. "
```
I don't think we've been very judicious about this in Nengo Loihi yet, but we should be using subclasses of `NengoException` in order to make it easy for other tools to catch / deal with exceptions coming from Nengo separate from other exceptions. In this case, I think I would make this a `BuildError`.

hunse replied:
Sounds good!

hunse commented:
Ok, I've addressed your comments, @tbekolay. I also added a commit that replaces all the generic exceptions with nengo ones (except one! see the commit comment).
